### ISSUE TITLE
ofPolyline no longer normalizes normals / tangents

### ISSUE THREAD

**Issue by bakercp** [openframeworks/openFrameworks#5288]
With the recent glm `ofPolyline` overhaul, the normals / tangents are no longer normalized.

A quick look makes it appear that they are being normalized here:

https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/graphics/ofPolyline.inl#L1014

and here:

https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/graphics/ofPolyline.inl#L1020

But ... it doesn't seem so in tests.

Test below.

```cpp
#pragma once


#include "ofMain.h"


class ofApp: public ofBaseApp
{
public:
    void setup() override
    {
        ofPolyline polyline;

        polyline.addVertex(ofRandomWidth(), ofRandomHeight(), 0);
        polyline.addVertex(ofRandomWidth(), ofRandomHeight(), 0);
        polyline.addVertex(ofRandomWidth(), ofRandomHeight(), 0);

        for (std::size_t i = 0; i < polyline.size(); ++i)
        {
            ofVec3f tangentPoint = polyline.getTangentAtIndex(i);
            std::cout << "Tangent " << i << " : " << tangentPoint << " / " << tangentPoint.normalize() << std::endl;
            ofVec3f normalPoint = polyline.getNormalAtIndex(i);
            std::cout << "Normal  " << i << " : " << normalPoint << " / " << normalPoint.normalize() << std::endl;
        }
    }


};
```

**Comment by tgfrerer** [comment]
Good find! That makes sense, as glm::normalize does not normalize in-place, but returns the normalized vector as a result. 

So in: https://github.com/openframeworks/openFrameworks/blob/master/libs/openFrameworks/graphics/ofPolyline.inl#L1014  
`glm::normalize(toGlm(tangent));`

I believe it should say: 

`tangent = glm::normalize(toGlm(tangent));`

Looking over the code in `ofPolyline`, there seem quite a few instances where the result of the normalization is never used.

**Comment by tgfrerer** [comment]
This one is a beauty to debug: windows will show the correct result, osx won't. The reason is less `ofPolyLine` (on which i'm still working) but the apparent evaluation order in the log stream operator. 

It appears that this line from the test: 
`		std::cout << "Tangent " << i << " : " << tangentPoint << " / " << tangentPoint.normalize() << std::endl;
`

* on windows, this first normalizes the tanget point inplace because of the call to `tangentPoint.normalize()`, then prints out the tangent point (well, the only just modified tangent point), so execution of the stream moves right to left, whereas 

* on OS X, this first prints out the tangent point value, then normalizes it in-place, and prints out the normalized value, so execution of the stream appears to move from left to right... 

One to keep in mind for automated tests (on windows it may appear everything is a-ok, while really it is not)!

**Comment by bakercp** [comment]
Nice analysis and good to know.  Is this something you'll take care of?  I'm happy to work on it, but it will be a few weeks before I can jump on it.  Thanks @tgfrerer loving this new glm stuff ...

**Comment by tgfrerer** [comment]
This should now be resolved via #5289, so i believe we can close this issue - @bakercp, maybe let me know should there still be a hiccup with the fixed version.
